### DISCUSSION TITLE
How do I import an ES module into my application using a native import map

### DISCUSSION THREAD

**Discussion by Aaronius** [webpack/webpack#17093] [Q&A]
I have an application being built using Webpack. I want to import a remote ES module using a native import map that's defined in my HTML page. From my application code, I'd like to import this ES module using an import statement:

```
import woof from "woof";
```

I do not want to do this:

```
const woof = async import('woof');
```

Is this possible? If so, how?

From reading the docs on `externalsType.module`, I see:

> Specify the default type of externals as 'module'. Webpack will generate code like import * as X from '...' for externals used in a module.

The term _module_ is generally vague (ES module specifically? Any type of module?), so I'm not sure if it applies to my situation. I also note that it says:

> Make sure to enable [experiments.outputModule](https://webpack.js.org/configuration/experiments/#experimentsoutputmodule) first, otherwise webpack will throw errors.

I'm not really trying to output my application as a module (I don't think so anyway), so this makes me feel like this is _not_ what I need.

Here's a repository that reflects basically what I'm trying to accomplish: https://github.com/Aaronius/webpack-import-map

Thanks for your help.

**Comment by alexander-akait** [comment]
webpack doesn't support HTML entrypoints currently, it is on our roadmap, you can have the `import` field in your `pacakge.json`

**Comment by Aaronius** [comment]
Thank you. Do you mind explaining what you mean by this and how it relates to what I'm trying to accomplish:

> you can have the import field in your `pacakge.json`

**Reply by alexander-akait** [reply]
webpack doesn't parse HTML, so webpack can't use `importmap` and can't undestand how to use inject `woof` module (anyway you can ignore module, and write how to load it), but you can use it in `package.json`:
```
{
  name: "",
  "imports": {
    "#woof": "./path/toYourModule.js"
  }
}
```
